### PR TITLE
fix: Remove viem's getAddress and use safeGetAddress instead

### PR DESCRIPTION
**Pull request by memoyil**
### Summary
🛠️🖼️🛡️

This pull request refactors and improves the address validation and checksumming logic across various files and components of the pancake-frontend app. It uses a common `isAddress` utility function from the `utils` module to ensure consistent and secure handling of addresses. It also handles some edge cases and errors related to fetching NFT data from the API and local storage.

> _Sing, O Muse, of the valiant code reviewers who refined the NFT market state_
> _With skillful use of `isAddress`, the handy helper from the `utils` module_
> _They checked and checksummed the addresses of tokens, collections, and contracts_
> _And handled gracefully the cases where the data was missing or corrupt_

### Walkthrough
*  Refactor the use of `getAddress` from `viem` to `isAddress` from `utils` for validating and checksumming addresses in various files ([link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-aedc95b3922e945e8e57ddcbadee3aed8a870eaca5cccd8aa241671272e76ab8L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-aedc95b3922e945e8e57ddcbadee3aed8a870eaca5cccd8aa241671272e76ab8L16-R16), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bL1-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bL17-R18), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bL29-R30), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a6c583ae9156b3647a7d3b110a6a5fdca92177e370cf7998ef39d136b4b6a73fL4-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a6c583ae9156b3647a7d3b110a6a5fdca92177e370cf7998ef39d136b4b6a73fL74-R99), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a6c583ae9156b3647a7d3b110a6a5fdca92177e370cf7998ef39d136b4b6a73fL103-R134), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-4b9d1d497762b8d67cf2e15af6da6f8cda35f6b3afee248bd96e6a7885825eefL34), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-4b9d1d497762b8d67cf2e15af6da6f8cda35f6b3afee248bd96e6a7885825eefR39), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-4b9d1d497762b8d67cf2e15af6da6f8cda35f6b3afee248bd96e6a7885825eefL139-R139), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a860f720209ae1f43227f2258b51d471b4739de55e3f7c033a0cd8d20fd98894L3), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a860f720209ae1f43227f2258b51d471b4739de55e3f7c033a0cd8d20fd98894L23-R27), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-ba3ce529911b349a2e11ee82dd74d3bc69f6cd09918fcf305cb9572567cd57acL11), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-ba3ce529911b349a2e11ee82dd74d3bc69f6cd09918fcf305cb9572567cd57acL40-R39), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-ba3ce529911b349a2e11ee82dd74d3bc69f6cd09918fcf305cb9572567cd57acL50-R50), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-b16df4570bba298265d9fa2c81b178e70d2cf09b484828f30c7b4ffb4e123642L1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-b16df4570bba298265d9fa2c81b178e70d2cf09b484828f30c7b4ffb4e123642L12-L19))
*  Make the `collectionAddress` parameter optional for various hooks and components that handle the case where the collection address is not defined or available, such as `useGetCollection`, `useCompleteNft`, `useGetCollectionDistribution`, `useNftOwn`, `IndividualNFTPage`, and `OneOfAKindNftPage` ([link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-5aa4b7186fcea0648595e2899011c59d2f19491226d05b223b09fa9e3017a19cL26-R26), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-9ad9a3edc9b341be9d7736c8f081406c378d39be9574c449e009738532f7e0b8L23-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-a860f720209ae1f43227f2258b51d471b4739de55e3f7c033a0cd8d20fd98894L23-R27), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-624c2ff826d5b9030dc4090abb4637228ef8b0a8874c3b0270a34861e9c0f19cL12-R12), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-624c2ff826d5b9030dc4090abb4637228ef8b0a8874c3b0270a34861e9c0f19cL53-R53), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-624c2ff826d5b9030dc4090abb4637228ef8b0a8874c3b0270a34861e9c0f19cL79-R79), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-5eb38b5e5808769e3c613fb4b8303e8a0f4bf66eca44c5888499e001c8c1a1c9L13-R13), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-5eb38b5e5808769e3c613fb4b8303e8a0f4bf66eca44c5888499e001c8c1a1c9L36-R36), [link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-ba3ce529911b349a2e11ee82dd74d3bc69f6cd09918fcf305cb9572567cd57acL40-R39))
*  Simplify the logic of the `useNftOwn` hook by removing the `let` variables and returning the object directly in each conditional branch, and by using the `nftIsProfilePic` variable consistently in each branch ([link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-624c2ff826d5b9030dc4090abb4637228ef8b0a8874c3b0270a34861e9c0f19cL27-R47))
*  Initialize the `apiResponse` variable of the `useGetCollectionDistributionPB` hook to `null`, to handle the case where the API call fails or returns null or undefined ([link](https://github.com/pancakeswap/pancake-frontend/pull/7984/files?diff=unified&w=0#diff-5eb38b5e5808769e3c613fb4b8303e8a0f4bf66eca44c5888499e001c8c1a1c9L36-R36))

### CODE REVIEW

File: apps/web/src/config/constants/priceHelperLps/pools/56.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { getAddress } from 'viem'
+import { isAddress } from 'utils'
```
Comment (chefjackson) [inline apps/web/src/config/constants/priceHelperLps/pools/56.ts:1]: can we rename it to getAddress? From the semantic meaning, it's getting the checksumed address

Comment (chefjackson) [reply apps/web/src/config/constants/priceHelperLps/pools/56.ts:1]: probably need to update the implementation as well. It's quite confusing to return both string and bool.

Can we return empty string or undefined to indicate it's not a valid address?

Comment (chefjackson) [reply apps/web/src/config/constants/priceHelperLps/pools/56.ts:1]: imo would be great to have two separate utility functions:
`isAddress(value: any): value is Address`
`getAddress(value: any): Address | undefined`

Comment (chef-jojo) [reply apps/web/src/config/constants/priceHelperLps/pools/56.ts:1]: safeGetAddress?
I think some of the place are meant to be throw error
Casting into false is just hiding the issue

**Review by chef-jojo — CHANGES_REQUESTED** [review]

File: apps/web/src/config/constants/priceHelperLps/pools/56.ts (hunk 1 above)
Comment (chefjackson) [reply apps/web/src/config/constants/priceHelperLps/pools/56.ts]: Great job! Will take a look

File: apps/web/src/utils/index.ts [hunk 2]
```diff
@@ -6,16 +6,16 @@ import memoize from 'lodash/memoize'
 import { TokenAddressMap } from '@pancakeswap/token-lists'
 import { chains } from './wagmi'
 
-// returns the checksummed address if the address is valid, otherwise returns false
-export const isAddress = memoize((value: any): `0x${string}` | false => {
+// returns the checksummed address if the address is valid, otherwise returns undefined
+export const safeGetAddress = memoize((value: any): `0x${string}` | undefined => {
```
Comment (chefjackson) [inline apps/web/src/utils/index.ts:10]: Can we change to return type to `Address | undefined`?

File: apps/web/src/utils/localStorageOrders.ts [hunk 3]
```diff
@@ -71,61 +71,67 @@ export function removeOrder(chainId: number, account: string, order: Order, pend
...
-      const ordersToSave = removeOrder(chainId, account, confirmedOrder)
-      ordersToSave.push({
-        ...confirmedOrder,
-        cancelledTxHash: cancelHash,
-      })
-      set(ordersKey, ordersToSave)
-    } else {
-      set(ordersKey, [
-        {
+  const checksummedAccount = safeGetAddress(account)
```
Comment (chefjackson) [inline apps/web/src/utils/localStorageOrders.ts:74]: previously it's throwing error right? Do we need to keep that behavior?

**Review by chefjackson — APPROVED** [review]

**Pull request merged**